storm-core/src/ui/public/topology.html

```diff
@@ -384,8 +389,51 @@ <h2 id="topology-resources-header">Topology resources</h2>
             $('#topology-configuration [data-toggle="tooltip"]').tooltip();
             $('#topology-actions [data-toggle="tooltip"]').tooltip();
             $('#topology-visualization [data-toggle="tooltip"]').tooltip();
+
+            var lagUrl = "/api/v1/topology/"+topologyId+"/lag";
+            $.getJSON(lagUrl,function(lagResponse,status,jqXHR) {
+              if (lagResponse !== null && lagResponse !== undefined && lagResponse instanceof Array && lagResponse.length > 0) {
+                var kafkaSpoutsLagTemplate = $(template).filter("#topology-kafka-spouts-lag-template").html();
+                var spoutsErrorTemplate = $(template).filter("#topology-spouts-lag-error-template").html();
+
+                var kafkaSpoutLags = lagResponse.filter(function(ele) {return ele.spoutType === "KAFKA";});
+                var isJson = function (input) {
+                  try {
+                    JSON.parse(input);
+                  } catch (e) {
+                    return false;
+                  }
+                  return true;
+                };
+                var kafkaSpoutsValidResults = kafkaSpoutLags.filter(function (ele) {return isJson(ele.spoutLagResult);});
+                var kafkaSpoutsErrorResults = kafkaSpoutLags.filter(function (ele) {return !isJson(ele.spoutLagResult);});
+                var data = {};
+                if (kafkaSpoutsValidResults.length > 0) {
+                  data.kafkaSpoutsLagResults = [];
+                  kafkaSpoutsValidResults.forEach(function(ele) {
+                    var spoutLagResult = JSON.parse(ele.spoutLagResult);
+                    spoutLagResult.forEach(function(ele2) {
```

Comment:
@abellina The spoutLagResult property is actually an array. Hence the flattening. 

Let me elaborate the limitations of valid and invalid approach. The TopologySpoutLag class in storm-core called by ui server does not know anything about if the spoutLagResult is valid or not. Reason is,  the way to get lag information for kafka and other spouts is handled by making a shell call handled by an external module since we did not want any direct dependency on storm core. For now we have kafka, but we can add other types of spouts as well. Plus the fields in ui or the template for a different type of spout could be different. The only commonality I found was if something went wrong for getting lag info for underlying spout(kafka or other) then an error message is sent in response. This will work for all types of spouts. To do what you are saying we will have to inspect the response in TopologySpoutLag class. I felt it did not matter if it were ui doing that or server. I preferred ui. Let me know if it makes sense or not and if you still think we need to change something.